pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -94,7 +94,18 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 	pendingTasks := map[api.JobID]*util.PriorityQueue{}
 
 	allNodes := util.GetNodeList(ssn.Nodes)
-
+	unlockedNodes := allNodes
```

Comment:
why not initialize it to empty here and remove latter `unlockedNodes = unlockedNodes[0:0]`